setup.py

```diff
@@ -46,6 +47,14 @@ def finalize_options(self):
 
 yubikey_require = ["yubikey-manager==4.0.*"]
 
+# Determine the appropriate version of pygit2 based on the Python version
+if sys.version_info >= (3, 9):
```

Comment:
Had to check something, but there was a reason why `pygit2` was pinned to `1.9`. A git security check was added in 1.9.1,  which broke something in our system. It'll need to stay pinned. Since our system is still on Python 3.10, we can set different versions of `pygit2` in case of 3.11 and 3.12 if that is necessary

Reply:
Sure, I can pin the version for `Python 3.10`. But the testcases used in pytest didn't raise any error when I ran the TAF builds with latest versions of `pygit2` for `Python 3.10`. On the other side `Python 3.11` needs `pygit2 version >=1.11.x` and `Python 3.12` needs `pygit2 version >=1.13.x` 

Should I just change it for  `Python 3.10` ?

Reply:
Let's pin it for `Python <= 3.10`. The error is not happening in TAF, it's in a component of our system which uses TAF. We'll update it if once we've addressed the core issue.